vendor/teleport-4.2/Makefile

```
@@ -0,0 +1,36 @@
+# Package details
+export VENDOR ?= gravitational
+export PACKAGE_REPO_NAME = teleport
+export MASTER_PACKAGE_NAME = teleport
+export MAJOR_VERSION = 4.2
+export PACKAGE_NAME = $(MASTER_PACKAGE_NAME)-$(MAJOR_VERSION)
+export PACKAGE_VERSION_PIN = $(MAJOR_VERSION).x
+
+export DEB_PACKAGE_ENABLED = true
+export APK_PACKAGE_ENABLED = false
+
+include ../../tasks/Makefile.package
+include ../../tasks/Makefile.apk
+include ../../tasks/Makefile.deb
+
+export DOWNLOAD_URL ?= https://get.gravitational.com/teleport/$(PACKAGE_VERSION)/teleport-v$(PACKAGE_VERSION)-$(OS)-$(ARCH)-bin.tar.gz
+# Starting with version 4.3, Teleport is not compatible with Alpine
+# so we disable updates
+# export AUTO_UPDATE_ENABLED = false
+export APK_BUILD_TEMPLATE ?= APKBUILD.github-binary
+export APKBUILD_DEPENDS += libc6-compat musl eudev-dev dpkg
+export PACKAGE_EXE = teleport tctl tsh
+export INSTALL_DIR = /usr/share/${MASTER_PACKAGE_NAME}/${MAJOR_VERSION}/bin
+
+install:
+	cd $(INSTALL_PATH) && $(CURL) -o - $(DOWNLOAD_URL) | tar --wildcards --strip-components=1 -zvx $(addprefix teleport/,$(PACKAGE_EXE))
+
+test:
+	teleport version | grep $(PACKAGE_VERSION)
+	tsh version | grep $(PACKAGE_VERSION)
+	tctl version | grep $(PACKAGE_VERSION)
```

We want to make sure that the `MAJOR_VERSION` for a version pinned package is correct. This is what saved us from accidentally advancing a package. 

```suggestion
	teleport version | grep $(MAJOR_VERSION)
	tsh version | grep $(MAJOR_VERSION)
	tctl version | grep $(MAJOR_VERSION)
```

---

I understand the concern, but would prefer to address it by checking the package version against the major version directly. 
```suggestion
	echo "$(PACKAGE_VERSION)" | grep "^$(MASTER_VERSION)"
	teleport version | grep $(PACKAGE_VERSION)
	tsh version | grep $(PACKAGE_VERSION)
	tctl version | grep $(PACKAGE_VERSION)
```

---

Ok - we can do that (note the other convention is used in all the other packages). Mind updating those then so we're consistent?

---

Done